spirv_glsl.cpp

```diff
@@ -725,44 +725,39 @@ void CompilerGLSL::emit_header()
 
 	if (!options.vulkan_semantics)
 	{
-		using Supp = KHR_shader_subgroupSupportHelper;
+		using Supp = ShaderSubgroupSupportHelper;
 
 		Supp::Result result = shader_subgroup_supporter.resolve();
 		for (uint32_t ft = 0u; ft < Supp::FeatureCount; ++ft)
 		{
 			Supp::Feature feature = static_cast<Supp::Feature>(ft);
-			if (!shader_subgroup_supporter.isFeatureRequested(feature))
+			if (!shader_subgroup_supporter.is_feature_requested(feature))
 				continue;
 
-			auto exts = shader_subgroup_supporter.getCandidatesForFeature(feature, result);
+			auto exts = shader_subgroup_supporter.get_candidates_for_feature(feature, result);
 			if (exts.empty())
 				continue;
 
-			statement("#ifdef GL_KHR_shader_subgroup");
-			statement("#extension GL_KHR_shader_subgroup : require");
 			for (auto it = exts.begin(); it != exts.end(); ++it)
 			{
-				std::string name = Supp::getExtensionName(*it);
-				auto extraNames = Supp::getExtraRequiredExtensionNames(*it);
-				if (extraNames.empty())
-					statement("#elif defined(", name, ")");
-				else
-				{
-					std::string extra;
-					for (const auto &e : extraNames)
-						extra += " && defined(" + e + ")";
-					statement("#elif defined(", name, ")", extra);
-				}
+				const Supp::Candidate ext = *it;
+
+				std::string name = Supp::get_extension_name(ext);
+				std::string extra = Supp::get_extra_required_extension_checks(ext);
+				auto extraNames = Supp::get_extra_required_extension_names(ext);
```

Comment:
I'd get the `extraPredicate` since right now you'll be requiring both AMD and NV extension.

its much easier to have
```glsl
#if defined(GL_VENDOR_extension_name) && ( predicate )
``` 
```

Reply:
ok now seeing the code, I can see that you've already done that

would be useful to call `extra` `extraPredicate` instead.